microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,314 @@
+| Array | Various internal lists, which show up in the devtools with the category name **(array)**. Like Hidden, this category clumps together a bunch of unrelated things. Many of the objects here are named **(object properties)** or **(object elements)**, indicating that they contain the string-keyed or numeric-keyed properties of a JavaScript object. |
+| Code | Things that grow proportionally to the amount of script, and/or the number of times that functions run. |
+| Synthetic | Synthetic nodes don't correspond to anything actually allocated in memory. These are used to distinguish the different kinds of GC roots. |
+
+### Edges
+
+Similar to the `nodes` array, the `edges` top-level array contains all the individual elements needed to recreate the edges of the graph.
+
+Also similar to nodes, the total number of edges can be calculated by multiplying `snapshot.edge_count` with `snapshot.meta.edge_fields.length`. Edges are also stored as a sequence of numbers which you will need to iterate on by groups of `snapshot.meta.edge_fields.length`.
+
+However, to read the `edges` array correctly, you first need read the `nodes` array because each node knows how many edges it has. 
+
+To recreate an edge you need three pieces of information:
+
+* The edge type.
+* The edge name or index.
+* The node the edge is connected to.
+
+You can use `snapshot.meta.edge_types` to understand what each element is.
+
+For example, if you read the first node in the `nodes` array and its `edge_count` property is set to 4, then the first 4 groups of `snapshot.meta.edge_fields.length` numbers in the `edges` array correspond to the 4 edges of this node.
+
+| Index in edge group | Name | Description |
+| --- | --- | --- |
+| `0` | `type` | The type of edge. See [Edge types](#edge-types) to find out what are the possible types. |
+| `1` | `name_or_index` | This can be a number or a string. If it’s a number, it corresponds to the index in the top-level `strings` array where the name of the edge can be found. |
+| `2` | `to_node` | The index within the `nodes` array that this edge is connected to. |
+
+### Edge types
+
+The first number found in the group of numbers for an edge in the `edges` array corresponds to its type. 
+
+This number is an index that can be used to lookup the type name in the `snapshot.meta.edge_types[0]` array.
+
+| Edge type | Description |
+| --- | --- |
+| Internal | Edges that don't correspond to JS-visible names but are still important. As an example, Function instances have a "context" representing the state of variables that were in scope where the function was defined. There is no way for JS code to directly read the "context" of a function, but it's still a very important link when trying to understand retainers. |
+| Weak | Weak edges don't keep the node they are connected to alive, and thus are omitted from the Retainers view. Any object with only weak edges pointing to it can be discarded by the GC. |
```

Comment:
```suggestion
| Weak | Weak edges don't keep the node they are connected to alive, and thus are omitted from the Retainers view. Any object with only weak edges pointing to it can be discarded by garbage collection (GC). |
```